applications/StructuralMechanicsApplication/python_scripts/element_sensitivity_domain_integration_process.py

```diff
@@ -0,0 +1,118 @@
+
+        self.element_sensitivity_variables = self.__generate_variable_list_from_input(parameter["element_sensitivity_variables"])
+
+
+    def ExecuteFinalizeSolutionStep(self):
+        """ This method is executed in order to finalize the current step
+
+        Here the dictionary containing the solution is filled
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        """
+        # loop over integration domains
+        for sub_mp_i in self.sensitivity_sub_model_parts:
+            # loop over sensitivty variables for which integration should performed
+            for variable_i in self.element_sensitivity_variables:
+                domain_sensitivity = 0.0
+                if self.__check_availability_of_sensitivities(variable_i, sub_mp_i):
+                    domain_sensitivity = KratosMultiphysics.VariableUtils().SumElementScalarVariable(variable_i, sub_mp_i)
+                    KratosMultiphysics.VariableUtils().SetNonHistoricalVariable(variable_i, domain_sensitivity, sub_mp_i.Elements )
+                else:
+                    raise Exception(variable_i.Name() + " is not available for domain integration!")
+
+
+    def __generate_variable_list_from_input(self, parameter):
+        """ Parse a list of variables from input.
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        parameter -- Kratos parameters containing process settings.
+        """
+
+        if not parameter.IsArray():
+            raise Exception("{0} Error: Variable list is unreadable".format(self.__class__.__name__))
+
+        variable_list = []
+        for i in range(0, parameter.size()):
+            variable_type = KratosMultiphysics.KratosGlobals.GetVariableType(parameter[i].GetString())
+            if (variable_type == "Double"):
+                variable_list.append(KratosMultiphysics.KratosGlobals.GetVariable( parameter[i].GetString() ))
+            else:
+                raise Exception("sensitivity domain integration is only available for variables of data type 'Double' but " + parameter[i].GetString() + " is of type '" + variable_type + "'.")
+
+        return variable_list
+
+
+    def __check_availability_of_sensitivities(self, variable, model_part):
+        """ Check if element sensitivities w.r.t. given variable are available.
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        variable -- traced variable within sensitivity analysis.
+        model_part -- sub model part of the sensitivity model part.
+        """
+
+        has_sensitivities = False
+        if sys.version_info[0] >= 3: # python3 syntax
+            has_sensitivities = model_part.Elements.__iter__().__next__().Has(variable)
+        else: # python2 syntax
+            has_sensitivities = model_part.Elements.__iter__().next().Has(variable)
+
+        return has_sensitivities
```

Comment:
the following is more comprehensive:
~~~py
if sys.version_info[0] >= 3: # python3 syntax
    return model_part.Elements.__iter__().__next__().Has(variable)
else: # python2 syntax
    return model_part.Elements.__iter__().next().Has(variable)
~~~